source/_integrations/matter.markdown

```diff
@@ -25,180 +25,207 @@ ha_integration_type: integration
+You do not yet have any Matter compatible hardware but you do like to try it out or maybe create your own DIY Matter device ? We have [prepared a page for you](https://nabucasa.github.io/matter-example-apps/) where you can easily flash Matter firmware to a supported ESP32 development board. We actually recommend the M5 Stamp C3 device running the Lightning app.
 
-### TP-Link Tapo P125M (power plug)
+NOTE for Android users: You need to follow the instructions at the bottom of the page to add the test device to the Google developer console, otherwise commissioning will fail. iOS users will not have this issue but they will get a prompt during commissioning asking if you trust the development device.
 
-This device runs Matter out of the box, so you can add it directly to the controller(s) of your choice!
+1. Make sure you are using the Google Chrome or Microsoft Edge browser.
+2. Open https://nabucasa.github.io/matter-example-apps/
+3. Attach the ESP32 device using an USB cable.
+4. Click the radiobutton next to the example you like to setup, in case of an M5 Stamp, click `Lightning app for M5STAMP C3`.
+5. Press `Connect`.
+6. In the popup dialog that appears choose the correct serial device, in most cases this will be something like "cu-usbserial" or alike.
+7. Click `Install Matter Lightning app example` and let it install the firmware on the device, this will take a few minutes.
+8. Once the device is flashed with the Matter firmware, connect to the device again but this time choose `Logs & console`.
+9. You are presented with a console interface where you see live logging of events. This is actually an interactive shell where you can type commands. For a list of all commands, type `matter help` and press enter.
+10. To add the device, we need the QR code. In the console type in `matter onboardingcodes ble` and copy/paste the URL in your browser.
+11. Use the QR code to add the device using one of the above instructions on your phone, e.g. using the Home Assistant Companion app.
 
-Look for the M addition in the model name, a device without the M (regular P125) is not Matter compliant. This device is available in the US only.
+## Known working devices
 
-[TP-Link Tapo P125M on Amazon](https://amzn.to/3RILJah)
+Because availability of actual Matter devices is sparse and proper HOWTO documentation even more, we provide a list of all known working devices that are tested by the Home Assistant Matter developers and/or the community. The list below is unordered and some of the links contain affiliate links.
 
-### ESP32 dev board running Matter example app
+Did you test a device that is not listed below ? It would be greatly appreciated if you share your experience either on the Matter discord channel or contibute a PR (or suggestion) to this documentation page so you can help out others, thanks in advance!
```

Comment:
```suggestion
Did you test a device that is not listed below ? It would be greatly appreciated if you share your experience either on the Matter discord channel or contribute a PR (or suggestion) to this documentation page so you can help out others, thanks in advance!
```